### PR TITLE
Fix compile error when define constant mode

### PR DESCRIPTION
**Pull request by lsh4711**
### Why need this chage? / Root cause:🔍
'undeclared' compilation error occurred when trying to use Mouse Key constant mode.
### Changes🔨
I fixed an error trying to compile options of accelerated(default) mode even when constant mode is defined.
### Screenshot📷
[QMK official docs]
SharedScreenshot7
SharedScreenshot8

(Options used in accelerated mode)
SharedScreenshot11

(Options used in constant mode)
SharedScreenshot12

[My config.h]
SharedScreenshot9

[Try compiling]
SharedScreenshot

[An example of changes]
SharedScreenshot10

[Compile success after changes and constant mode works very well] :) 👍 
SharedScreenshot2

### CODE REVIEW

**Comment by lsh4711** [comment]
Admin who is reading this message, please check my pull request..

File: quantum/qmk_settings.c [hunk 1]
```diff
@@ -27,16 +27,18 @@ static void auto_shift_timeout_apply(void) {
...
+        mk_delay = QS.mousekey_delay / 10;
+        mk_interval = QS.mousekey_interval;
+        mk_max_speed = QS.mousekey_max_speed;
+        mk_time_to_max = QS.mousekey_time_to_max;
+        mk_wheel_delay = QS.mousekey_wheel_delay / 10;
+        mk_wheel_interval    = QS.mousekey_wheel_interval;
+        mk_wheel_max_speed   = QS.mousekey_wheel_max_speed;
+        mk_wheel_time_to_max = QS.mousekey_wheel_time_to_max;
+    }
+#    endif
```
Comment (xyzz) [inline quantum/qmk_settings.c]: Change to `#if defined(MOUSEKEY_ENABLE) && !defined(MK_3_SPEED)`, remove whitespace change

File: quantum/qmk_settings.c [hunk 2]
```diff
@@ -49,15 +51,17 @@ static const qmk_settings_proto_t protos[] PROGMEM = {
...
-   DECLARE_STATIC_SETTING_NOTIFY(9, mousekey_delay, mousekey_apply),
-   DECLARE_STATIC_SETTING_NOTIFY(10, mousekey_interval, mousekey_apply),
-   DECLARE_STATIC_SETTING_NOTIFY(11, mousekey_move_delta, mousekey_apply),
-   DECLARE_STATIC_SETTING_NOTIFY(12, mousekey_max_speed, mousekey_apply),
-   DECLARE_STATIC_SETTING_NOTIFY(13, mousekey_time_to_max, mousekey_apply),
-   DECLARE_STATIC_SETTING_NOTIFY(14, mousekey_wheel_delay, mousekey_apply),
-   DECLARE_STATIC_SETTING_NOTIFY(15, mousekey_wheel_interval, mousekey_apply),
-   DECLARE_STATIC_SETTING_NOTIFY(16, mousekey_wheel_max_speed, mousekey_apply),
-   DECLARE_STATIC_SETTING_NOTIFY(17, mousekey_wheel_time_to_max, mousekey_apply),
+#    ifndef MK_3_SPEED
```
Comment (xyzz) [inline quantum/qmk_settings.c]: Same as above

File: quantum/qmk_settings.c [hunk 3]
```diff
@@ -167,15 +171,17 @@ void qmk_settings_reset(void) {
...
-    QS.mousekey_delay = MOUSEKEY_DELAY;
-    QS.mousekey_interval = MOUSEKEY_INTERVAL;
-    QS.mousekey_move_delta = MOUSEKEY_MOVE_DELTA;
-    QS.mousekey_max_speed = MOUSEKEY_MAX_SPEED;
-    QS.mousekey_time_to_max = MOUSEKEY_TIME_TO_MAX;
-    QS.mousekey_wheel_delay = MOUSEKEY_WHEEL_DELAY;
-    QS.mousekey_wheel_interval = MOUSEKEY_WHEEL_INTERVAL;
-    QS.mousekey_wheel_max_speed = MOUSEKEY_WHEEL_MAX_SPEED;
-    QS.mousekey_wheel_time_to_max = MOUSEKEY_WHEEL_TIME_TO_MAX;
+#    ifndef MK_3_SPEED
```
Comment (xyzz) [inline quantum/qmk_settings.c]: Same as above

**Review by xyzz — CHANGES_REQUESTED** [review]

**Comment by lsh4711** [comment]
Thank you so much! @xyzz 
Vial build completed, and everything is working fine. :)

File: quantum/qmk_settings.c [hunk 4]
```diff
@@ -48,16 +48,16 @@ static const qmk_settings_proto_t protos[] PROGMEM = {
...
+    DECLARE_STATIC_SETTING_NOTIFY(9, mousekey_delay, mousekey_apply),
+    DECLARE_STATIC_SETTING_NOTIFY(10, mousekey_interval, mousekey_apply),
+    DECLARE_STATIC_SETTING_NOTIFY(11, mousekey_move_delta, mousekey_apply),
+    DECLARE_STATIC_SETTING_NOTIFY(12, mousekey_max_speed, mousekey_apply),
+    DECLARE_STATIC_SETTING_NOTIFY(13, mousekey_time_to_max, mousekey_apply),
+    DECLARE_STATIC_SETTING_NOTIFY(14, mousekey_wheel_delay, mousekey_apply),
+    DECLARE_STATIC_SETTING_NOTIFY(15, mousekey_wheel_interval, mousekey_apply),
+    DECLARE_STATIC_SETTING_NOTIFY(16, mousekey_wheel_max_speed, mousekey_apply),
+    DECLARE_STATIC_SETTING_NOTIFY(17, mousekey_wheel_time_to_max, mousekey_apply),
 #endif
```
Comment (xyzz) [inline quantum/qmk_settings.c]: There's still whitespace change here that needs to be removed

**Review by xyzz — CHANGES_REQUESTED** [review]

File: quantum/qmk_settings.c (hunk 4 above)
Comment (lsh4711) [reply quantum/qmk_settings.c]: Oh.. i will do it right away.

**Pull request merged**